components/movementsensor/adxl345/adxl345.go

```diff
@@ -41,9 +41,64 @@ var modelName = resource.NewDefaultModel("accel-adxl345")
 
 // AttrConfig is a description of how to find an ADXL345 accelerometer on the robot.
 type AttrConfig struct {
-	BoardName              string `json:"board"`
-	I2cBus                 string `json:"i2c_bus"`
-	UseAlternateI2CAddress bool   `json:"use_alternate_i2c_address,omitempty"`
+	BoardName              string              `json:"board"`
+	I2cBus                 string              `json:"i2c_bus"`
+	UseAlternateI2CAddress bool                `json:"use_alternate_i2c_address,omitempty"`
+	SingleTap              *TapAttrConfig      `json:"tap,omitempty"`
+	FreeFall               *FreeFallAttrConfig `json:"free_fall,omitempty"`
+}
+
+// TapAttrConfig is a description of the configs for tap registers.
+type TapAttrConfig struct {
+	AccelerometerPin int     `json:"accelerometer_pin"`
+	InterruptPin     string  `json:"interrupt_pin"`
+	ExcludeX         bool    `json:"exclude_x,omitempty"`
+	ExcludeY         bool    `json:"exclude_y,omitempty"`
+	ExcludeZ         bool    `json:"exclude_z,omitempty"`
+	Threshold        float32 `json:"threshold,omitempty"`
+	Dur              float32 `json:"dur_us,omitempty"`
+}
+
+// FreeFallAttrConfig is a description of the configs for free fall registers.
+type FreeFallAttrConfig struct {
+	AccelerometerPin int     `json:"accelerometer_pin"`
+	InterruptPin     string  `json:"interrupt_pin"`
+	Threshold        float32 `json:"threshold,omitempty"`
+	Time             float32 `json:"time_ms,omitempty"`
+}
+
+func (tapCfg *TapAttrConfig) ValidateTapConfigs(path string) error {
+	if tapCfg.AccelerometerPin != 1 && tapCfg.AccelerometerPin != 2 {
+		return errors.New("adxl345: Accelerometer pin must be 1 or 2")
+	}
+	if tapCfg.Threshold != 0 {
+		if tapCfg.Threshold < 0 || tapCfg.Threshold > 15937 {
+			return errors.New("adxl345: Tap threshold must be 0 between and 15,937mg")
+		}
+	}
+	if tapCfg.Dur != 0 {
+		if tapCfg.Dur < 0 || tapCfg.Dur > 159375 {
+			return errors.New("adxl345: Tap dur must be between 0 and 160,000µs")
+		}
+	}
+	return nil
+}
+
+func (freefallCfg *FreeFallAttrConfig) ValidateFreeFallConfigs(path string) error {
+	if freefallCfg.AccelerometerPin != 1 && freefallCfg.AccelerometerPin != 2 {
+		return errors.New("Accelerometer pin must be 1 or 2")
```

Comment:
nit: some errors have "adxl345:" some don't. e.g. this one doesn't, tap configvalidation does.